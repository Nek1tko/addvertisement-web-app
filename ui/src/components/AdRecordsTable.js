import React from "react";
import { Box } from "@material-ui/core";
import { DataGrid } from '@mui/x-data-grid';
import eren from '../img/eren.jpg'

var id = 0

function createData(image, name, location, price) {
    id += 1
    return { id, image, name, location, price };
}

const rows = [
    createData(eren, 'Танк', 'Улица Дыбенко', 10000000),
    createData(eren, 'Стирательная резинка', 'Проспект Большевиков', 10),
    createData(eren, 'Кот', 'Сенная', 0),
];

const columns = [
    {
        field: 'image',
        headerName: 'Превью',
        minWidth: 100,
        renderCell: (params) => <img
            src={params.value}
            style={{ height: '100%', width: '100%', objectFit: 'contain' }}
        />
    },
    { field: 'name', headerName: 'Название', minWidth: 400, sortable: false },
    { field: 'location', headerName: 'Метро', minWidth: 200, sortable: false },
    { field: 'price', headerName: 'Цена', width: 100, sortable: false },
];

export default function AdRecordsTable() {
    const [page, setPage] = React.useState(0);

    return (
        <Box>
            <DataGrid
                rowHeight={100}
                autoHeight
                page={page}
                onPageChange={(newPage) => setPage(newPage)}
                pageSize={10}
                columns={columns}
                rows={rows}
                pagination
                {...rows}
                disableColumnMenu
            />
        </ Box>
    );
}
