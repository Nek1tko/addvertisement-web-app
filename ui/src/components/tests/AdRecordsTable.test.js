/**
 * @group unit
 */

import { mount, shallow } from 'enzyme';
import AdRecordsTable from '../AdRecordsTable';
import { createBrowserHistory } from 'history';
import { act } from 'react-dom/test-utils';

const mCategory = {
    id: 1,
    name: "mock category"
};

const mSubcategory = {
    id: 1,
    name: "mock subcategory",
    category: mCategory
};

const mMetro = {
    id: 1,
    name: "Улица Дыбенко",
};

const mUserId = 1;

const mAds = [
    {
        id: 1,
        description: "mock description",
        price: "123",
        name: "mock name",
        metro: mMetro,
        subCategory: mSubcategory,
        isActive: true,
        isFavourite: true,
        saler: { id: mUserId }
    },
    {
        id: 2,
        description: "mock description 2",
        price: "1234",
        name: "mock name 2",
        metro: mMetro,
        subCategory: mSubcategory,
        isActive: true,
        isFavourite: true,
        saler: { id: mUserId + 1 }
    },
];

describe('AdRecordsTableTests', () => {
    let mHistory, wrapper, props

    beforeEach(() => {
        mHistory = {
            ...createBrowserHistory(),
            push: jest.fn()
        };
        props = {
            history: mHistory,
            userId: mUserId,
            ads: mAds
        }
        wrapper = mount(<AdRecordsTable {...props} />);
    })

    afterEach(() => {
        jest.clearAllMocks();
    })

    test('AdRecordTableTest', () => {
        mount(<AdRecordsTable {...props} />);
    });

    test('AdRecordTableSnapshotTest', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('AdRecordTableDataGridTest', () => {
        const dataGrid = wrapper.find('#adDataGrid').first();

        let mClick = {
            row: {
                saler: {
                    id: mUserId
                }
            }
        }
        act(() => {
            dataGrid.props().onRowClick(mClick);
        });

        expect(mHistory.push).toHaveBeenCalledWith({ ad: mClick.row, pathname: "/seller-ad" });

        mClick.row.saler.id = mUserId + 1;
        act(() => {
            dataGrid.props().onRowClick(mClick);
        });

        expect(mHistory.push).toHaveBeenCalledWith({ ad: mClick.row, pathname: "/customer-ad" });
    });
});
