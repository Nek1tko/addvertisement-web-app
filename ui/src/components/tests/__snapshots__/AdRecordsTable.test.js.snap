// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdRecordsTableTests AdRecordTableSnapshotTest 1`] = `ReactWrapper {}`;
